app/src/main/java/com/project/gamedb/base/BaseAdapter.kt

```diff
@@ -16,4 +16,12 @@ abstract class BaseAdapter<T> : RecyclerView.Adapter<BaseViewHolder<T>>() {
 
     override fun getItemCount(): Int = items.size
 
+
+    fun replaceData(collection: List<T>?) {
+        collection?.let {
+            items.clear()
+            items.addAll(it)
+        }
+        notifyDataSetChanged()
+    }
```

Comment:
1. getItemId dùng khi nào nhỉ? 
2. Hàm notifyDataSetChanged() được gọi thì recycler view sẽ hoạt động như nào nhỉ? 
2. Nếu replaceData(null) thì recyclerview sẽ thay đổi như nào nhỉ?

Reply:
1. getItemId dùng khi muốn lấy ID của item tại vị trí cố định của item trong list
2. Hàm notifyDataSetChanged() được gọi thì recycler view sẽ reload lại toàn bộ item 
3. Nếu replaceData(null) thì recyclerview sẽ không thay đổi vì list truyền và đã được check null

Reply:
2. reload thì cụ thể nó sẽ thực hiện gì với mỗi item nhỉ? 
3. null hay không thì nó vẫn notifyDataSetChanged đó mà nhỉ

Reply:
2. Nó sẽ clear toàn bộ item trong list cũ và bindView lại từ đầu với list mới
3.  notifyDataSetChanged đc gọi và thực hiện quá trình clear rồi bind lại, nhưng do list k bị thay đổi nên dữ liệu nhìn thấy vẫn như cũ